File: List/even_odd.py
class node:
    def __init__(self, data):
        self.data = data
        self.next = None


class ListNode:

    def __init__(self):
        self.head = None

    def insert(self, val):
        if self.head is None:
            self.head = node(val)
        else:
            temp = self.head

            while temp.next is not None:
                temp = temp.next

            temp.next = node(val)

    def display(self):
        temp = self.head

        while temp is not None:
            print(temp.data)
            temp = temp.next

    def find_length(self):
        fast = self.head
        slow = self.head

        while fast and fast.next:
            fast=fast.next.next
        
        if fast is None:
            return 'Even Length'
        else:
            return 'Odd Length'

        


if __name__ == "__main__":

    l = ListNode()

    for i in range(4):
        i+=1
        l.insert(i)

    l.display()

    print(l.find_length())
    

File: List/even_odd.cpp
#include<bits/stdc++.h>
using namespace std;

struct node{
    int data;
    struct node *next;
};

typedef struct node *N;

N getnode(int x){
    N t=(struct node *)malloc(sizeof(struct node));
    t->data = x;
    t->next = NULL;
    return t;
}

N insert(N start,int x){
    N t = getnode(x);

    if(!start){
        return t;
    }

    N c = start;

    while(c->next!=NULL){
        c=c->next;
    }

    c->next = t;
    return start;
}

void display(N start){
    N t=start;

    while(t){
        cout << t->data << endl;
        t=t->next;
    }

}

string find_length(N start){
    N fast=start;
    N slow= start;

    while(fast and fast->next){
        fast=fast->next->next;
    }
    if(!fast){
        return "even";
    }
    return "odd";
}

int main(){
    N start = NULL;

    for(int i=0;i<5;i++){
        start = insert(start, i);
    }

    display(start);


    cout << "Length is " << find_length(start) << endl;  

    return 0;
}

